lib/rails/observers/railtie.rb

```diff
@@ -20,6 +18,7 @@ class Railtie < ::Rails::Railtie
       end
 
       config.after_initialize do |app|
+        require 'active_record/base'
```

Comment:
Yes. Adding this line means that we are running all `on_load(:active_record)` right here. If there are circular dependency warning I think it is being caused by one of these blocks.